### PR TITLE
Add  CHiME4 recipe

### PR DESCRIPTION
**Pull request by funcwj**
Add CHiME4 recipe with character unit. Training data combines WSJ0+1 & CHiME4 real & simulation utterances. Current best WER with CTC prefix beam search are 21.17%/19.06%/28.41%/29.16% on dt05 real/simu and et05 real/simu, respectively.

### CODE REVIEW

File: wenet/dataset/dataset.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def Dataset(data_type, data_list_file, symbol_table, conf,
     else:
         dataset = Processor(dataset, processor.parse_raw)
 
-    dataset = Processor(dataset, processor.tokenize, symbol_table, bpe_model)
+    dataset = Processor(dataset, processor.tokenize, symbol_table, bpe_model, conf.get('char', False))
```
Comment (robin1001) [inline wenet/dataset/dataset.py:142]: the line is too long.

File: wenet/dataset/processor.py [hunk 2]
```diff
@@ -258,7 +258,7 @@ def compute_fbank(data,
         yield dict(key=sample['key'], label=sample['label'], feat=mat)
 
 
-def tokenize(data, symbol_table, bpe_model=None):
+def tokenize(data, symbol_table, bpe_model=None, char=False):
```
Comment (robin1001) [inline wenet/dataset/processor.py:261]: tokenize use char as the default model unit. 
You mean your model unit is seperated with white space? it is more like phoneme, such as timit. It's better we take another name.

**Review by robin1001 — CHANGES_REQUESTED** [review]
please see inline.

File: examples/chime4/s0/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+## Conformer Result
+
+* Feature info: dither + specaug + speed perturb
+* Training info: lr 0.0005, batch size 8, 1 gpu, acc_grad 4, 80 epochs
+* Decoding info: average_num 10
+
+|   decoding mode | dt05_real_1ch | dt05_simu_1ch | et05_real_1ch | et05_simu_1ch |
+|:---------------:|:-------------:|:-------------:|:-------------:|:-------------:|
+| ctc_beam_search |   19.06%      |   21.17%      |   28.39%      |    29.16%     |
+|  att_rescoring  |   17.92%      |   20.22%      |   27.40%      |    28.25%     |
```
Comment (robin1001) [inline examples/chime4/s0/README.md:12]: ctc_beam_search -> ctc_prefix_beam_search
att_rescoring       ->  attention_rescoring
rename so it is consistent with other recipes.

File: wenet/dataset/processor.py (hunk 2 above)
Comment (funcwj) [reply wenet/dataset/processor.py:261]: I'd like to use split_with_space instead

**Review by robin1001 — APPROVED** [review]

**Pull request merged**